proposals/008-simplifying-imports-to-improve-compilation-performance.md

```diff
@@ -0,0 +1,71 @@
+# Simplifying imports within GOV.UK Frontend to improve compilation performance
+
+**Closing date:** 1 May 2020
+
+**Status:** Proposed
+
+## Summary
+
+**This is an iteration of [proposal #7](https://github.com/alphagov/govuk-design-system-architecture/pull/21), based on the feedback we received from that proposal.**
+
+We’re making changes to the way that the base layers (settings, tools and helpers) are imported within different parts of GOV.UK Frontend’s Sass, to reduce the time it takes to compile to CSS.
+
+Most of you will not be affected by these changes, and if you import "node_modules/govuk-frontend/govuk/all" you will not need to do anything to benefit from these performance improvements.
+
+It will have the most impact on you if you import only specific parts of GOV.UK Frontend.
+
+## Problem
+
+It can take a long time to compile GOV.UK Frontend’s Sass to CSS. This is because many of the individual files within GOV.UK Frontend import their own dependencies, even if they would usually have already been imported. We originally did this so that you could import specific bits of GOV.UK Frontend without having to separately import their dependencies.
```

Comment:
> We originally did this so that you could import specific bits of GOV.UK Frontend without having to separately import their dependencies.

I'm on the opinion that we should continue to value the principle of importing specific bits without having to separately import dependencies. The fact that the legacy sass implementation (where 'modularisation' was only possible with `@import` and `@export`) didn't allow any sort of 'import-once'/'tree-shaking' was overcome later on with `@use` which allows us to preserve this principle.